docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc

```diff
@@ -0,0 +1,60 @@
+== DOM updates should not lead to cross-site scripting (XSS) attacks
+
+=== Policy Details
+
+[width=45%]
+[cols="1,1"]
+|=== 
+|Prisma Cloud Policy ID 
+| 9743b4fb-4e52-484e-bc51-e8700e9f99aa
+
+|Checkov ID 
+|CKV3_SAST_196
+
+|Severity
+|HIGH
+
+|Subtype
+|Build
+
+|Language
+|JavaScript, TypeScript
+
+|CWEs
+|https://cwe.mitre.org/data/definitions/79.html[CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')]
+
+|OWASP Categories
+|https://owasp.org/Top10/A03_2021-Injection/[A03:2021 - Injection]
+
+|=== 
+
+=== Description
+
+This policy detects unsafe DOM updates that can lead to Cross-Site Scripting (XSS) attacks in JavaScript and TypeScript applications. XSS vulnerabilities occur when untrusted data is written to the DOM without proper validation or sanitization, allowing attackers to inject malicious scripts.
+
+Vulnerable code example:
+
+[source,JavaScript]
+----
+const userInput = document.location.search;
+document.write(userInput);
+document.getElementById('output').innerHTML = userInput;
+----
+
+In this code snippet, user input from `document.location.search` is directly written to the DOM using `document.write` and `innerHTML` without any validation or sanitization. This exposes the application to XSS attacks, where an attacker could inject malicious scripts through the user input.
+
+=== Fix - Buildtime
+
+To fix this issue, ensure that user input is properly sanitized before being written to the DOM. Use libraries such as DOMPurify to sanitize user input and prevent XSS attacks.
```

Comment:
Ensure that user input is properly sanitized before being written to the DOM. Use libraries such as DOMPurify to sanitize user input and prevent XSS attacks.